src/main/java/org/prebid/server/bidder/BidderCatalog.java

```diff
@@ -49,67 +52,89 @@ private void validateBidderName(String bidderName) {
 
     private Map<String, String> createErrorsForDeprecatedNames(BidderInstanceDeps deps) {
         return deps.getDeprecatedNames().stream().collect(Collectors.toMap(
-                Function.identity(),
+                String::toLowerCase,
                 deprecatedName -> ERROR_MESSAGE_TEMPLATE_FOR_DEPRECATED.formatted(deprecatedName, deps.getName())));
     }
 
     private void processVendorId(BidderInstanceDeps coreDeps, String bidderName) {
-        final BidderInfo bidderInfo = coreDeps.getBidderInfo();
-        final BidderInfo.GdprInfo gdprInfo = bidderInfo != null ? bidderInfo.getGdpr() : null;
-        final Integer vendorId = gdprInfo != null ? gdprInfo.getVendorId() : null;
-
-        if (vendorId != null && vendorId != 0) {
-            vendorIdToBidderName.put(vendorId, bidderName);
-        }
+        Optional.ofNullable(coreDeps.getBidderInfo())
+                .map(BidderInfo::getGdpr)
+                .map(BidderInfo.GdprInfo::getVendorId)
+                .filter(vendorId -> vendorId != 0)
+                .ifPresent(vendorId -> vendorIdToBidderName.put(vendorId, bidderName));
     }
 
     /**
      * Returns a list of registered bidder names.
      */
     public Set<String> names() {
-        return new HashSet<>(bidderDepsMap.keySet());
+        return Collections.unmodifiableSet(biddersNames);
     }
 
     /**
      * Tells if given name corresponds to any of the registered bidders.
      */
     public boolean isValidName(String name) {
-        return bidderDepsMap.containsKey(name);
+        return Optional.ofNullable(name)
+                .map(String::toLowerCase)
+                .map(bidderDepsMap::containsKey)
```

Comment:
Maybe we need to update bidderDepsMap to be caseInsensitive, than no toLowerCase needed for this class